src/document/document.lisp

```diff
@@ -472,6 +472,46 @@
 (defvar *html-subformat* nil)
 (defvar *document-tight* nil)
 
+(defvar *pandoc-program* "pandoc")
+
+(defvar *pandoc-options* nil)
+
+;; Trying to represent YAML as an alist is rife with corner cases, so
+;; for now just store the block as a string.
+;; Allowing an alist will be backwards compatible, so leave the decision
+;; for later.
+(defvar *pandoc-metadata-block* "")
+
+;; FIXME Better function name?
+;; FIXME Use :markdown output, it's simpler, and adjust at the end,
+;; so use something like pandoc-subformat, or markdown-subformat.
+;; FIXME If defvar values contain \\ like #\ chars, need to escape.
+;; Maybe in a tex-escape function, texencode or whatever.
+;; FIXME The MGL-PAX documentation has two leading spaces in front
+;; of every docstring, which causes Pandoc to misinterpret code blocks
+;; and probably various other things.
+;; Ah, that's probably what is causing issues in some lists.
+(defun print-output (parse-tree stream &key format)
+  (case format
+    (:pandoc-pdf
+     (let ((proc (uiop:launch-program (append (list *pandoc-program*)
+                                              *pandoc-options*
+                                              (list "-f" "markdown"
+                                                    "-t" "pdf"
+                                                    "-o" "-"))
+                                      :input :stream
+                                      :output stream
+                                      :error-output t)))
```

Comment:
The UIOP bundled with current SBCL seems to be buggy, I had to specify `:error-output *error-output*` explicitly, else it would complain about `t`.